plugins/woocommerce/includes/class-wc-install.php

```diff
@@ -1163,6 +1167,133 @@ public static function create_terms() {
 		}
 	}
 
+	/**
+	 * Install and activate the WooCommerce Legacy REST API plugin from the WordPress.org directory if all the following is true:
+	 *
+	 * 1. We are in a WooCommerce upgrade process (not a new install).
+	 * 2. The 'woocommerce_skip_legacy_rest_api_plugin_auto_install' filter returns false (which is the default).
+	 * 3. The plugin is not installed and active already (but see note about multisite below).
+	 * 4. The Legacy REST API is enabled in the site OR the site has at least one webhook defined that uses the Legacy REST API payload format (disabled webhooks also count).
+	 *
+	 * In multisite setups it could happen that the plugin was installed by an installation process performed in another site.
+	 * In this case we check if the plugin was autoinstalled in such a way, and if so we activate it if the conditions are fulfilled.
+	 */
+	private static function maybe_install_legacy_api_plugin() {
+		if ( self::is_new_install() ) {
+			return;
+		}
+
+		/**
+		 * Filter to skip the automatic installation of the WooCommerce Legacy REST API plugin
+		 * from the WordPress.org plugins directory.
+		 *
+		 * @since 8.8.0
+		 *
+		 * @param bool $skip_auto_install False, defaulting to "don't skip the plugin automatic installation".
+		 * @returns bool True to skip the plugin automatic installation, false to install the plugin if necessary.
+		 */
+		if ( apply_filters( 'woocommerce_skip_legacy_rest_api_plugin_auto_install', false ) ) {
+			return;
+		}
+
+		if ( ( 'yes' !== get_option( 'woocommerce_api_enabled' ) &&
+			0 === wc_get_container()->get( Automattic\WooCommerce\Internal\Utilities\WebhookUtil::class )->get_legacy_webhooks_count( true ) ) ) {
+			return;
+		}
+
+		$plugin_name = 'woocommerce-legacy-rest-api/woocommerce-legacy-rest-api.php';
+
+		wp_clean_plugins_cache();
+		if ( isset( get_plugins()[ $plugin_name ] ) ) {
+			if ( ! ( get_site_option( 'woocommerce_autoinstalled_plugins', array() )[ $plugin_name ] ?? null ) ) {
+				// The plugin was installed manually so let's not interfere.
+				return;
+			}
+
+			if ( in_array( $plugin_name, wp_get_active_and_valid_plugins(), true ) ) {
+				return;
+			}
+
+			// The plugin was automatically installed in a different installation process - can happen in multisite.
+			$install_ok = true;
+		} else {
+			$install_result = wc_get_container()->get( PluginInstaller::class )->install_plugin(
+				'https://downloads.wordpress.org/plugin/woocommerce-legacy-rest-api.1.0.1.zip',
```

Comment:
🤔 Too bad there doesn't seem to be a URL like `https://downloads.wordpress.org/plugin/woocommerce-legacy-rest-api.latest.zip` available to use here. You could do something like the following to retrieve the latest download link rather than hard coding it:

```php
require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
$info = plugins_api( 'plugin_information', array( 'slug' => 'woocommerce-legacy-rest-api' ) )
$downloadlink = $info->download_link;
```

Reply:
Actually, it looks like the correct URL is `https://downloads.wordpress.org/plugin/woocommerce-legacy-rest-api.latest-stable.zip`